components/post_view/post_list_virtualized/index.jsx

```diff
@@ -33,21 +33,37 @@ export function makeCountUnreadsBelow() {
     );
 }
 
+/* This connected component is written mainly for maintaining the unread count to be passed to the toast
+   Unread count logic:
+   * If channel is at the latest set of posts:
+      Unread count is the Number of posts below new message line
+   * if channel is not at the latest set of posts:
+      1. UnreadCount + any recent messages in the latest chunk.
+      2. If channel was marked as unread.
+        * Unread count of channel alone.
+*/
+
 function makeMapStateToProps() {
     const countUnreadsBelow = makeCountUnreadsBelow();
     const preparePostIdsForPostList = makePreparePostIdsForPostList();
-
-    return (state, ownProps) => {
-        let postIds = getPostIdsInChannel(state, ownProps.channelId);
-
-        if (postIds) {
-            postIds = preparePostIdsForPostList(state, {postIds, lastViewedAt: ownProps.lastViewedBottom, channelId: ownProps.channelId});
+    return function mapStateToProps(state, ownProps) {
+        let newRecentMessagesCount = 0;
+        const channelMarkedAsUnread = isManuallyUnread(state, ownProps.channelId);
+        if (!ownProps.atLatestPost) {
+            const lastViewedAt = state.views.channel.lastChannelViewTime[ownProps.channelId];
+            let postIds = getPostIdsInChannel(state, ownProps.channelId);
+            if (postIds) {
+                postIds = preparePostIdsForPostList(state, {postIds, lastViewedAt, channelId: ownProps.channelId});
+            }
+            newRecentMessagesCount = countUnreadsBelow(state, postIds, lastViewedAt);
```

Comment:
if `postIds` is falsy, how will behave the selector? if we can ensure is an empty array, then it's fine, but there are other falsy values, maybe we should act a little defensive here.
```suggestion
            newRecentMessagesCount = countUnreadsBelow(state, (postIds || []), lastViewedAt);
```
or maybe, knowing that it is empty, have a function that returns earlier?

Reply:
Selector has a check for 

```
            if (!postIds) {
                return 0;
            }
```

Wouldn't that suffice?